client/my-sites/email/email-providers-comparison/index.jsx

```diff
@@ -119,11 +139,11 @@ class EmailProvidersComparison extends React.Component {
 	};
 
 	goToEmailForwarding = () => {
-		const { domain, currentRoute, selectedSiteSlug } = this.props;
+		const { currentRoute, selectedDomainName, selectedSiteSlug } = this.props;
 
 		recordTracksEvent( 'calypso_email_providers_add_click', { provider: 'email-forwarding' } );
 
-		page( emailManagementForwarding( selectedSiteSlug, domain.name, currentRoute ) );
+		page( emailManagementForwarding( selectedSiteSlug, selectedDomainName, currentRoute ) );
```

Comment:
I'm using `selectedDomainName` instead of `domain.name` whenever I can in this component as we are sure the former is always available.